src/binder_voicecall.c

```diff
@@ -1257,28 +1383,61 @@ binder_voicecall_supp_svc_notification(
     GBinderReader reader;
 
     gbinder_reader_copy(&reader, args);
-    ssn = gbinder_reader_read_hidl_struct(&reader, RadioSuppSvcNotification);
-    if (ssn) {
-        if (ssn->isMT) {
+    if (self->interface_aidl == RADIO_AIDL_INTERFACE_NONE) {
+        ssn = gbinder_reader_read_hidl_struct(&reader, RadioSuppSvcNotification);
+        if (ssn) {
+            if (ssn->isMT) {
+                struct ofono_phone_number ph;
+
+                /* MT unsolicited result code */
+                DBG_(self, "MT code: %d, index: %d type: %d number: %s", ssn->code,
+                     ssn->index, ssn->type, ssn->number.data.str);
+                if (ssn->number.data.str && ssn->number.len) {
+                    ph.type = ssn->type;
+                    g_strlcpy(ph.number, ssn->number.data.str, sizeof(ph.number));
+                } else {
+                    ph.type = OFONO_NUMBER_TYPE_UNKNOWN;
+                    ph.number[0] = 0;
+                }
+
+                ofono_voicecall_ssn_mt_notify(self->vc, 0, ssn->code,
+                    ssn->index, &ph);
+            } else {
+                /* MO intermediate result code */
+                ofono_voicecall_ssn_mo_notify(self->vc, 0, ssn->code, ssn->index);
+            }
+        }
+    } else {
+        gboolean is_mt;
+        gint32 code, index, type;
```

Comment:
In some places these get initialized before use, here not. Really needed or not? Does the gbinder function always set a value or not?